scripts/database-maintenance.sh

```
@@ -0,0 +1,151 @@
+# Main execution
+main() {
+    log "=== Starting database maintenance ==="
+    log "Database directory: $DB_DIR"
+
+    if [[ ! -d "$DB_DIR" ]]; then
+        log "ERROR: Database directory $DB_DIR not found"
+        exit 1
+    fi
+
+    # Find all SQLite databases
+    local databases=($(find "$DB_DIR" -name "*.db" -type f 2>/dev/null))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)